test/model/src/include/model/util.h

```diff
@@ -91,28 +130,91 @@ class wiredtiger_session_guard {
 };
 
 /*
- * wt_cursor_get_string --
- *     Search in WiredTiger using the provided cursor. Return a string result, or NONE if not found.
- *     Throw an exception on error.
+ * config_map --
+ *     A configuration map.
  */
-inline data_value
-wt_cursor_get_string(WT_CURSOR *cursor, const data_value &key)
-{
-    const char *s;
-    int ret;
+class config_map {
+    using value_t = std::variant<std::string, std::shared_ptr<config_map>>;
 
-    set_wt_cursor_key(cursor, key);
-    ret = cursor->search(cursor);
-    if (ret != 0) {
-        if (ret == WT_NOTFOUND)
-            return NONE;
-        throw wiredtiger_exception(cursor->session, ret);
+public:
+    /*
+     * config_map::from_string --
+     *     Parse config map from a string.
+     */
+    static config_map from_string(const char *str, const char **end = NULL);
+
+    /*
+     * config_map::from_string --
+     *     Parse config map from a string.
+     */
+    static inline config_map
+    from_string(const std::string &str)
+    {
+        return from_string(str.c_str());
     }
 
-    ret = cursor->get_value(cursor, &s);
-    if (ret != 0)
-        throw wiredtiger_exception(cursor->session, ret);
-    return data_value(s);
+    /*
+     * config_map::contains --
+     *     Check whether the config map contains the given key.
+     */
+    inline bool
+    contains(const char *key) const noexcept
+    {
+        return _map.find(key) != _map.end();
+    }
+
+    /*
+     * config_map::get_map --
+     *     Get the corresponding config map value. Throw an exception on error.
+     */
+    inline std::shared_ptr<config_map> const
+    get_map(const char *key)
+    {
+        return std::get<std::shared_ptr<config_map>>(_map.find(key)->second);
+    }
+
+    /*
+     * config_map::get_string --
+     *     Get the corresponding string value. Throw an exception on error.
+     */
+    inline std::string
+    get_string(const char *key) const
+    {
+        return std::get<std::string>(_map.find(key)->second);
+    }
+
+    /*
+     * config_map::get_uint64 --
+     *     Get the corresponding integer value. Throw an exception on error.
+     */
+    inline uint64_t
+    get_uint64(const char *key) const
+    {
+        std::istringstream stream(std::get<std::string>(_map.find(key)->second));
+        uint64_t v;
+        stream >> v;
+        return v;
+    }
+
+private:
+    /*
+     * config_map::config_map --
+     *     Create a new instance of the config map.
+     */
+    inline config_map() noexcept {};
```

Comment:
It's more like I didn't have a need to make it public—as the object instance is immutable, I didn't feel like there was a need to create an empty `config_map` that can't be further modified.

But it also wouldn't hurt to make it public; that would allow you to statically declare a `config_map` variable and then assign this to it—that would be the only real use case.

Reply:
All good - it's fine as-is, I just wondered why.